pkg/kclient/all.go

```diff
@@ -26,19 +26,24 @@ func (c *Client) GetAllResourcesFromSelector(selector string, ns string) ([]unst
 }
 
 func getAllResources(client dynamic.Interface, apis []apiResource, ns string, selector string) ([]unstructured.Unstructured, error) {
-	var mu sync.Mutex
 	var wg sync.WaitGroup
 	var out []unstructured.Unstructured
+	outChan := make(chan []unstructured.Unstructured)
```

Comment:
I would probably create a new function instead of modifying this generic one: this optimization is valid for logs, probably not for other routines using it.

Reply:
> I would probably create a new function instead of modifying this generic one: this optimization is valid for logs, probably not for other routines using it.


I made this optimization with two things in mind:
1. mutex vs channel
2. all go routines trying to append to a single slice
 
Instead of locking and unlocking every time, I am putting the received slice on the channel and using it in `append` call later. We can do without a mutex because it's an unbuffered channel, and hence only one go routine can put `[]unstructured.Unstructured` to it at a time. So, I think this optimization should be generalized because it's made purely from Go perspective without a thought about which functionality of odo could benefit from it.

Do you see a downside to this optimization being generalized? If yes, I would surely like to understand what it is, and try to optimize this only from `odo logs` perspective instead of generalizing it.

Reply:
You are right, I misread your changes in this function